### DISCUSSION TITLE
when adding a validator to the model, the route starts accepting any body

### DISCUSSION THREAD

**Discussion by Prof1-web** [fastapi/fastapi#8370] [Questions] [answered]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, validator


class User(BaseModel):
    name: str

    @validator("name")
    def validate(cls, v):
        print(v)
        return v


app = FastAPI()


@app.post("/user")
async def user(u: User):
    User(name="john")
    return u
```

### Description

Send post request to `/user` with this body: `"{"email": "test@gmail.com"}"`. Route return 200 OK. But this json does not match the User model, route must return 422 http error code. Route accepts any body ignoring my model.

validator is not working properly. When I send a request with body `"{"name": "john"}"` the print function should write "john" to the console. But she writes `{'name': 'john'}` instead. The problem is not related to pydantic. Inside the route, another User model is created. Everything is fine with it and the print function outputs what it should

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.79.0

### Python Version

3.10

**Comment by iudeen** [comment, marked as answer]
Issue seems happen because of your function name inside the `BaseModel`. 

`validate` is a class member function of  `BaseModel` and that is what FastAPI uses to validate your request payloads. 

`validate` is overriding some inner function calls by FastAPI. 

Changing your definition to something like this fixes the issue. 
```python
class User(BaseModel):
    name: str

    @validator("name")
    def custom_validate(cls, v):
        print(v)
        return v

```

**Comment by Prof1-web** [comment]
> Issue seems happen because of your function name inside the `BaseModel`.
> 
> `validate` is a class member function of `BaseModel` and that is what FastAPI uses to validate your request payloads.
> 
> `validate` is overriding some inner function calls by FastAPI.
> 
> Changing your definition to something like this fixes the issue.
> 
> ```python
> class User(BaseModel):
>     name: str
> 
>     @validator("name")
>     def custom_validate(cls, v):
>         print(v)
>         return v
> ```

It's right. I changed function name and it worked. Thank you
